packages/headers-parser/src/normalize.ts

```diff
@@ -4,39 +4,71 @@ import type { Mapper } from 'map-obj'
 
 import { getForRegExp } from './for_regexp.js'
 import { splitResults } from './results.js'
-import type { Header } from './types.js'
+import type { Header, MinimalHeader } from './types.js'
+
+export interface MinimalNormalizedHeaders {
+  headers: MinimalHeader[]
+  errors: Error[]
+}
+
+export interface NormalizedHeaders {
+  headers: Header[]
+  errors: Error[]
+}
 
 // Validate and normalize an array of `headers` objects.
 // This step is performed after `headers` have been parsed from either
 // `netlify.toml` or `_headers`.
-export const normalizeHeaders = function (headers: any, minimal: boolean) {
+export function normalizeHeaders(headers: MinimalHeader[], minimal: true): MinimalNormalizedHeaders
+export function normalizeHeaders(headers: MinimalHeader[], minimal: false): NormalizedHeaders
+export function normalizeHeaders(
+  headers: MinimalHeader[],
+  minimal: boolean,
+): MinimalNormalizedHeaders | NormalizedHeaders
+export function normalizeHeaders(
+  headers: MinimalHeader[],
+  minimal: boolean,
+): MinimalNormalizedHeaders | NormalizedHeaders {
   if (!Array.isArray(headers)) {
     const error = new TypeError(`Headers must be an array not: ${headers}`)
-    return splitResults([error])
+    return splitResults<never>([error])
```

Comment:
This looks funny but is correct: `splitResults` takes an array of `T | Error` and returns `{values: T[]: errors: Error[]}`; when given a literal array of type `Error[]` it can't infer `T`, so I explicitly pass in `never` as `T`.

Reply:
Can you add this as an inline comment?